Lib/inspect.py

```diff
@@ -968,20 +973,27 @@ def getabsfile(object, _filename=None):
 
 modulesbyfile = {}
 _filesbymodname = {}
+_moduleless = set()
 
 def getmodule(object, _filename=None):
     """Return the module an object was defined in, or None if not found."""
     if ismodule(object):
         return object
     if hasattr(object, '__module__'):
         return sys.modules.get(object.__module__)
+    code = _getcode(object)
+    if code and hash(code) in _moduleless:
```

Comment:
The `hash(code)` is calculated multiple times in this method. Can you calculate it only once and re-use the result?